include/boost/geometry/algorithms/detail/sub_range.hpp

```diff
@@ -44,7 +49,7 @@ struct sub_range<Geometry, Tag, false>
 template <typename Geometry>
 struct sub_range<Geometry, polygon_tag, false>
 {
-    typedef typename geometry::ring_type<Geometry>::type & return_type;
+    typedef typename geometry::ring_type<Geometry>::type const& return_type;
```

Comment:
Well in my code `ring_type<>::type` is mutable while range::at() returns a const. My ring type is `iterator_range<const point *>`. Let me investigate futher...